product_combination_exclude/models/product_product.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = 'product.product'
+
+    @api.multi
+    def create(self, vals):
+        vals = dict(vals or {})
+        if 'attribute_value_ids' in vals:
+            exclusion_obj = self.env['product.attribute.exclude']
+            search_args = []
+            if 'product_tmpl_id' in vals:
+                search_args.extend([
+                    '|',
+                    ('product_tmpl_ids', 'in', [vals['product_tmpl_id']]),
+                    ('product_tmpl_ids', 'in', [])])
+            try:
+                search_args.extend([
+                    ('attribute_value_ids', 'in',
+                     vals['attribute_value_ids'][0][2])
+                ])
+                attr_vals = self.env['product.attribute.value'].browse(
```

Comment:
We could slightly reduce complexity by moving these few lines outside of the try/except, which would then explicitly guard only for the above `IndexError`. After that, the exception is not needed because we've validated the index & the subsequent line doesn't use it. Returning in the except already cuts off our program flow for the bad index.